app/screens/clock_display/clock_display.android.js

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+    Platform,
+    TouchableOpacity,
+    Modal,
+    View
+} from 'react-native';
+
+import StatusBar from 'app/components/status_bar';
+import FormattedText from 'app/components/formatted_text';
+import RadioButtonGroup from 'app/components/radio_button';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import ClockDisplayBase from './clock_display_base';
+
+export default class ClockDisplay extends ClockDisplayBase {
+    static propTypes = {
+        showModal: PropTypes.bool.isRequired,
+        close: PropTypes.func.isRequired
+    };
+
+    setMilitaryTime = (value) => {
+        this.setState({
+            militaryTime: value
+        });
+    };
+
+    closeModal = () => {
+        const {militaryTime, close} = this.props;
+        this.setState({
```

Comment:
if you are closing the modal why do we set the state? and more so equal to a prop

Reply:
this.props.militaryTime is used to reset the militaryTime from state.  Maybe a better name for this.props.militaryTime is this.props.defaultMilitaryTIme

Reply:
I've changed this.state.militaryTime to this.state.newMilitaryTime.  Seemed like a better change than this.props.defaultMilitaryTime